index.js

```diff
@@ -1,62 +1,61 @@
 /* eslint-env node */
 'use strict';
-
-const resolve = require('resolve');
-
-function debugMacros(features) {
-  let plugins = [
-    [
-      require.resolve('babel-plugin-debug-macros'),
-      {
-        flags: [
-          {
-            source: 'ember-m3/feature-flags',
-            flags: features,
-          },
-        ],
-      },
-      '@ember-data/canary-features-stripping',
-    ],
-  ];
-
-  return plugins;
-}
-
-function enabledFeatures(isDevelopingAddon) {
-  let features = {
-    CUSTOM_MODEL_CLASS: false,
-  };
-  try {
-    let emberDataPath = require.resolve('ember-data');
-    let buildInfra = resolve.sync('@ember-data/private-build-infra/src/features', {
-      cwd: emberDataPath,
-    });
-    let emberDataFeatures = require(buildInfra)(process.env.EMBER_ENV === 'production');
-    if ('CUSTOM_MODEL_CLASS' in emberDataFeatures) {
-      features.CUSTOM_MODEL_CLASS = emberDataFeatures.CUSTOM_MODEL_CLASS;
-    }
-  } catch (e) {
-    features = { CUSTOM_MODEL_CLASS: isDevelopingAddon ? null : false };
-  }
-
-  return features;
+const Funnel = require('broccoli-funnel');
+const getDebugMacros = require('./src/debug-macros').debugMacros;
+
+/*
+ shouldIncludeChildAddon runs during
+ addon init so does not have access to
+ the host application instance.
+*/
+function hasDataPackage(addon) {
+  let addons = addon.project.addonPackages;
+  return addons['ember-data'] !== undefined;
 }
 
 module.exports = {
   name: 'ember-m3',
 
   included() {
-    this._super.included.apply(this, arguments);
+    this._super.included.call(this, ...arguments);
+    this.configureBabelOptions();
+  },
+
+  shouldIncludeChildAddon(addon) {
+    if (!hasDataPackage(this)) {
+      return true;
+    }
```

Comment:
Isn't it more complicated than this?

If a user has their own `@ember-data/store` shouldn't that trump the  version we implicitly bring along?

Reply:
@hjdivad we could make it more complicated in the future if we want. For now I don't think I would recommend it. Detecting is difficult enough as is.

Reply:
Added this detection